File: Muhindhar_Pytest_Assessment_POM/Pytest_POM/pages/searchpage.py
from selenium.webdriver.common.by import By

class SearchPage:

    search = (By.XPATH, "//input[@placeholder='Search']")
    sbtn = (By.XPATH, "//button[contains(@class,'btn-default')]")
    check = (By.XPATH, "//div[@id='content']//h1")

File: Muhindhar_Playwright_Assessment/pages/searchpage.ts
import {Page, Locator } from '@playwright/test'

export class Searchpage{
    readonly page:Page;
    readonly searchbar : Locator;
    readonly clksearch:Locator;

    constructor(page:Page){
        this.page = page;
        this.searchbar = page.locator("//input[@placeholder='Search']");
        this.clksearch=page.locator("//button[@class='btn btn-default btn-lg']");
}

    async seacrpro(product:string){
        await this.searchbar.fill(product);
        await this.clksearch.click();
    }

}